### PR TITLE
Clean up to finish RC1

### PR DESCRIPTION
**Pull request by craig-a-swanson**
# Description
Various small bug fixes and cleaning of code. Added email link functionality (a RC2 item)

Fixes # (issue)
Finalizes for RC1 sign-off

## Type of change

Please delete options that are not relevant.

- [x ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)

## Change Status

- [x ] Complete, tested, ready to review and merge

# How Has This Been Tested?

- [x ] numerous user tests on different devices

# Checklist

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x ] There are no merge conflicts

### CODE REVIEW

File: DJHelper/DJHelper/EventPageViewController.swift [hunk 1]
```diff
@@ -43,20 +62,55 @@ class EventPageViewController: UIViewController {
...
         addSongButtonProperties.colorTheme()
     }
-    /*
-    // MARK: - Navigation
 
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    // Provides feature to send a URL via the Mail app.
+    // When testing, it only works on a device, not the simulator.
+    private func setupEmailForLink() {
+        guard MFMailComposeViewController.canSendMail(),
```
Comment (mikeflow22) [inline DJHelper/DJHelper/EventPageViewController.swift:74]: I've never heard of this class - it's good to be exposed to new things! Did you look at UIActivityViewController? I think if we are unhappy with the MFMailComposeVC we can use the activityVC as a backup.

File: DJHelper/DJHelper/EventPageViewController.swift [hunk 2]
```diff
@@ -68,3 +122,27 @@ extension EventPageViewController: UITableViewDataSource, UITableViewDelegate {
...
+        }
+
+        switch result {
+
+        case .cancelled:
+            self.dismiss(animated: true, completion: nil)
+        case .saved:
+            return
+        case .sent:
+            self.dismiss(animated: true, completion: nil)
```
Comment (mikeflow22) [inline DJHelper/DJHelper/EventPageViewController.swift:141]: What are your thoughts on us providing an alert here saying it was successfully sent and then dismissing the VC?

File: DJHelper/DJHelper/View Controllers/CreateEventViewController.swift [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 import UIKit
 
-class CreateEventViewController: UIViewController {
+class CreateEventViewController: ShiftableViewController {
```
Comment (mikeflow22) [inline DJHelper/DJHelper/View Controllers/CreateEventViewController.swift:11]: Great use of inheritance.

**Review by mikeflow22 — APPROVED** [review]

**Comment by craig-a-swanson** [comment]
I’ve never used UIActivityVuewController and I’m open to all options. If that’s better then I’m all for it.

The reason I didn’t present an alert saying the message was sent was because in the Apple documentation it said the app has no way of knowing if the message was successfully sent. Like if there was no signal and the message couldn’t be sent or something like that. Probably an unlikely scenario though. 



Sent with my mobile device.

**Pull request merged**